zpeclient_test.go

```diff
@@ -522,10 +609,57 @@ func TestAuthorize(t *testing.T) {
 		Resource: "home.domain:pods",
 	}
 	res, err = privateCache.authorize("fakeclient", check)
+	if err != nil {
+		t.Error(err)
+	}
 	if res {
 		t.Error("Wrong authorization result, fakeclient's request should be denied")
 	}
 
+	// test case: one policy has two assertions, first assertion with explicitly Allow,
+	// second assertion with explicitly Deny.
+	check = AthenzAccessCheck{
+		Action:   "delete",
+		Resource: "home.domain.deny:services",
+	}
+	res, err = privateCache.authorize(username, check)
+	if err != nil {
+		t.Error(err)
+	}
+	if res {
+		t.Error("Wrong authorization result, username's request should be denied since assertion has an explicit DENY")
+	}
+
+	check = AthenzAccessCheck{
+		Action:   "create",
+		Resource: "home.domain.deny:services",
+	}
+	res, err = privateCache.authorize(username, check)
+	if err != nil {
+		t.Error(err)
+	}
+	if !res {
+		t.Error("Wrong authorization result, username's request should be allowed")
+	}
+
+	check = AthenzAccessCheck{
+		Action:   "delete",
+		Resource: "domain.does.not.exist:services",
+	}
+	res, err = privateCache.authorize(username, check)
+	if err == nil {
+		t.Error("there should be error because such domain doesn't exist in the cache")
+	}
+
+	check = AthenzAccessCheck{
+		Action:   "delete",
+		Resource: "domain.wrong.format.services",
+	}
+	res, err = privateCache.authorize(username, check)
+	if err == nil {
```

Comment:
Could you also check if the error string matches exactly the one expected to make sure we're getting the right error back?